### PR TITLE
New Datasource: azurerm_virtual_hub_connection

### PR DESCRIPTION
**Pull request by TechyMatt**

### CODE REVIEW

File: internal/services/network/virtual_hub_connection_data_source.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+
+									"address_prefixes": {
+										Type:     pluginsdk.TypeSet,
```
Comment (tombuildsstuff) [inline internal/services/network/virtual_hub_connection_data_source.go:100]: these'll need to be a List in a data source, else they all get the same hash value and combined into a single item:

```suggestion
										Type:     pluginsdk.TypeList,
```

File: internal/services/network/virtual_hub_connection_data_source.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+							Computed: true,
+						},
+
+						"propagated_route_table": {
+							Type:     pluginsdk.TypeList,
+							Computed: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"labels": {
+										Type:     pluginsdk.TypeSet,
```
Comment (tombuildsstuff) [inline internal/services/network/virtual_hub_connection_data_source.go:70]: (as below) in a data source this'll need to be a TypeList

```suggestion
										Type:     pluginsdk.TypeList,
```

File: website/docs/d/virtual_hub_connection.html.markdown [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - The name of the Virtual Hub Connection.
+
+* `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.
+*  `virtual_hub_name` - The name which should be used for Virtual hub using the connection.
```
Comment (tombuildsstuff) [inline website/docs/d/virtual_hub_connection.html.markdown:34]: can we make this:

```suggestion
*  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.
```

File: website/docs/d/virtual_hub_connection.html.markdown [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+output "virtual_hub_connection_id" {
+  value = data.azurerm_virtual_hub_connection.example.id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - The name of the Virtual Hub Connection.
```
Comment (tombuildsstuff) [inline website/docs/d/virtual_hub_connection.html.markdown:31]:
```suggestion
* `name` - The name of the Connection which should be retrieved.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
A few minor comments but this otherwise LGTM - thanks for this @TechyMatt

**Comment by TechyMatt** [comment]
Thanks @tombuildsstuff changes made per comments.

File: website/docs/d/virtual_hub_connection.html.markdown [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - The name of the Connection which should be retrieved.
+
+* `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.
+*  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.
```
Comment (katbyte) [inline website/docs/d/virtual_hub_connection.html.markdown:34]:
```suggestion
* `resource_group_name` - The Name of the Resource Group where the Virtual Hub Connection exists.

*  `virtual_hub_name` - The name of the Virtual Hub where this Connection exists.
```

**Review by katbyte — APPROVED** [review]
Thanks @TechyMatt - aside from one minor comment LGTM 🔧

**Pull request merged**